runtime-sdk/modules/contracts/src/store.rs

```diff
@@ -2,17 +2,36 @@
 use oasis_contract_sdk_types::storage::StoreKind;
 use oasis_runtime_sdk::{
     context::Context,
+    keymanager::{self, StateKey},
     storage::{self, Store},
 };
 
 use crate::{state, types, Error, MODULE_NAME};
 
+/// Confidential store key pair ID domain separation context base.
+pub const CONFIDENTIAL_STORE_KEY_PAIR_ID_CONTEXT_BASE: &[u8] =
+    b"oasis-runtime-sdk/contracts: state";
+
 /// Create a contract instance store.
 pub fn for_instance<'a, C: Context>(
     ctx: &'a mut C,
     instance_info: &types::Instance,
     store_kind: StoreKind,
-) -> Result<impl Store + 'a, Error> {
+) -> Result<Box<dyn Store + 'a>, Error> {
+    let confidential_key: Option<StateKey> = if let StoreKind::Confidential = store_kind {
+        let kmgr_client = ctx.key_manager().ok_or(Error::Unsupported)?;
+        let kid = keymanager::get_key_pair_id(&[
+            CONFIDENTIAL_STORE_KEY_PAIR_ID_CONTEXT_BASE,
+            &instance_info.id.to_storage_key(),
+        ]);
+        let kp = kmgr_client
+            .get_or_create_keys(kid)
+            .map_err(|err| Error::ExecutionFailed(err.into()))?;
```

Comment:
This should result in a batch abort (e.g. define a new error variant that has the `sdk_error(abort)` attribute and use that).